File: ruby/add_big_numbers.rb
def add_big_numbers(n1, n2)
  n1 = n1.split('')
  n2 = n2.split('')
  sum = ''
  carry = 0
  while (n1.any? || n2.any? || carry > 0) 
    carry += n1.pop.to_i + n2.pop.to_i
    sum = (carry % 10).to_s + sum
    carry = carry > 9 ? 1 : 0
  end

  sum
end

p add_big_numbers('11', '22')
p add_big_numbers('111111111111111111111111111111111111111111111111111', '22')

File: js/addBigNumbers.js

// case 1: same length 
console.log(addBigNumbers('1', '2'));
// case 2: different length
console.log(addBigNumbers('1111111', '2'));

// case 3: long nums
console.log(addBigNumbers('111111111111111111111111111111111111111111111', '2222'));

function addBigNumbers(n1, n2) {
  var n1Split = n1.split('');
  var n2Split = n2.split('');
  var sum = '';
  var carry = 0;
  while ( n1Split.length || n2Split.length || carry ) {
    carry += ~~n1Split.pop() + ~~n2Split.pop();
    sum = carry % 10 + sum;
    carry = carry > 9;
  }
  
  return sum;
}

